src/http/headers.cr

```diff
@@ -57,24 +57,24 @@ struct HTTP::Headers
     @hash = Hash(Key, String | Array(String)).new
   end
 
-  def []=(key, value : String)
+  def []=(key : String | HTTP::Headers::Key, value : String) : String
```

Comment:
`HTTP::Headers::Key` is an internal type. It seems odd that it would be exposed in a public method. Any idea where this union is coming from? Most similar methods only receive `String` key.

Reply:
Hm, I'm not sure where this reference came from :/ Removing `HTTP::Headers::Key` from here didn't raise any exceptions :shrug: 

For my own education, what makes 'HTTP::Headers::Key` internal, beyond being defined as a child of `HTTP::Headers`? Should it be private in this case?

Reply:
It's `nodoc`